In [2]:
import tensorflow as tf
from tensorflow import keras
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from keras.models import Sequential
from keras.layers import Dense, Dropout, Activation
from keras.optimizers import SGD


from keras.utils import to_categorical

Using TensorFlow backend.


In [3]:
#data = pd.read_csv ('/Users/Tim/Documents/GitHub/NBA-Draft-Biola-Summer-Research/finished_joseph_csv_manual_pie - joseph_csv_manual_pie.csv')
data = pd.read_csv ('/Users/josephhartono/Downloads/finished_joseph_csv_manual_pie - joseph_csv_manual_pie.csv')
data.drop('PLAYER_ID', inplace=True, axis=1)
data.drop('FIRST_NAME_x', inplace=True, axis=1)
data.drop('LAST_NAME_x', inplace=True, axis=1)
data.drop('PLAYER_NAME_x', inplace=True, axis=1)
data.drop('HEIGHT_WO_SHOES_FT_IN', inplace=True, axis=1)
data.drop('WINGSPAN_FT_IN', inplace=True, axis=1)
data.drop('STANDING_REACH_FT_IN', inplace=True, axis=1)
data.drop('DRAFT_CLASS', inplace=True, axis=1)
data.drop('SEASON_ID', inplace=True, axis=1)
data.drop('GP', inplace=True, axis=1)
data.drop('GS', inplace=True, axis=1)
data.drop('MIN', inplace=True, axis=1)
data.drop('ORGANIZATION_ID', inplace=True, axis=1)
data.drop('SCHOOL_NAME', inplace=True, axis=1)
data.drop('PLAYER_AGE', inplace=True, axis=1)
data.drop('LEAGUE_ID', inplace=True, axis=1)
#PG SG SF PF C
mapping = {'PG':1,'PG-SG':1,'PG-SF':1,'PG-PF':1,'PG-C':1,
           'SG':2,'SG-PG':2,'SG-SF':2,'SG-PF':2,'SG-C':2,
           'SF':3,'SF-PG':3,'SF-SG':3,'SF-PF':3,'SF-C':3,
           'PF':4,'PF-PG':4,'PF-SG':4,'PF-SF':4,'PF-C':4,
           'C':5,'C-PG':5,'C-SG':5,'C-SF':5,'C-PF':5}
data = data.replace({'POSITION_x':mapping})

#data['PIE_score_5YR'] = pd.qcut(data['PIE_score_5YR'], q=[0, .2, .4, .6, .8, 1], labels = [10 ,20, 30,40,50])
print(type(data["PIE_score_5YR"][0]))
#print(data.columns)
data

<class 'numpy.float64'>


POSITION_x  HEIGHT_WO_SHOES  WEIGHT  WINGSPAN  STANDING_REACH  FGM   FGA  \
0           1.0            76.50   175.0     82.00           102.5  6.2  15.0   
1           3.0            77.50   214.0     81.50           103.5  5.4  11.7   
2           5.0            83.75   266.0     90.50           113.5  4.7   7.9   
3           3.0            79.25   223.0     84.00           103.0  4.2   8.8   
4           3.0            78.75   226.0     81.00           105.0  5.4  11.5   
..          ...              ...     ...       ...             ...  ...   ...   
194         4.0            82.25   259.8     88.50           112.5  6.3  12.5   
195         1.0            71.50   184.0     75.50            91.5  7.7  18.0   
196         4.0            79.25   248.4     85.50           108.0  5.9  10.0   
197         4.0            78.50   223.8     86.75           106.5  4.2   7.2   
198         3.0            79.00   227.8     83.25           101.5  6.0  13.7   

     FG_PCT  FG3M  FG3A  ...  OREB  DREB   REB  AST  STL  BLK  TOV   PF   PTS  \
0     0.412   1.9   5.9  ...   0.4   2.4   2.8  4.5  1.1  0.9  3.1  1.1  16.6   
1     0.464   0.2   0.7  ...   3.6   3.7   7.3  1.2  2.0  0.3  2.1  2.7  14.6   
2     0.592   0.0   0.0  ...   3.2   4.2   7.3  1.3  0.5  3.6  2.0  2.9  12.3   
3     0.479   0.9   2.6  ...   1.7   3.8   5.4  2.7  0.8  0.8  3.0  2.8  11.3   
4     0.468   1.2   2.6  ...   2.2   4.2   6.4  2.6  1.4  0.4  2.3  1.5  14.2   
..      ...   ...   ...  ...   ...   ...   ...  ...  ...  ...  ...  ...   ...   
194   0.505   0.9   2.4  ...   2.9   7.4  10.3  1.6  0.5  1.4  1.7  2.8  17.1   
195   0.428   1.8   5.5  ...   1.3   4.1   5.4  4.5  1.9  0.2  2.3  1.4  23.5   
196   0.595   1.1   1.9  ...   2.8   5.5   8.3  1.1  1.0  0.7  2.6  2.8  19.5   
197   0.577   0.1   0.1  ...   2.6   4.8   7.4  1.0  0.9  1.8  1.9  3.1  12.0   
198   0.440   1.3   3.6  ...   1.9   3.3   5.2  1.1  1.1  0.3  1.9  1.8  17.1   

     PIE_score_5YR  
0              9.7  
1              0.0  
2              5.0  
3             14.5  
4             12.5  
..             ...  
194           14.0  
195           13.5  
196           12.7  
197            8.3  
198           11.2  

[199 rows x 24 columns]

In [11]:
#data

#splits the data
train_x = data.sample(frac = 0.7)
validation_x = data.sample(frac = 0.3)

#gets pie score of train_x
#gets pie score of validation_x
train_y = train_x["PIE_score_5YR"].values
train_y = np.asarray(train_y)

validation_y = validation_x["PIE_score_5YR"].values
validation_y = np.asarray(validation_y)


# display(train_y)
# display(validation_y)

#takes out pie score from train_x
#takes out pie score from validation_x
train_x.drop('PIE_score_5YR', inplace=True, axis=1)
validation_x.drop('PIE_score_5YR', inplace=True, axis=1)


#just gets the values of train_x without the headers
train_x = train_x.values
validation_x = validation_x.values

# train_y = np.asarray(train_y)
display(validation_y)




display(validation_y)

# validation_y = validation_y[:,None]
# train_y = train_y[:,None]
# display(validation_y)
# display(train_y)

# train_y = to_categorical(train_y, 5)
# validation_y = to_categorical(validation_y, 5)

# display(train_x)
# display(test_x)






array([ 8.7, 12.2, 10.4, 12.7, 12. , 10.6,  6.5,  6.1,  9. , 15.6,  0. ,
        0. ,  9.5,  8.1,  0. ,  9.9, 10.3, 14.1,  9.3, 10.1, 13.5, 11. ,
       11.4, 10.5,  6.8, 12.2,  0. , 13.8,  7.4,  0. ,  0. ,  0. , 12.9,
        0. ,  7.3,  8.2,  0. ,  8.8,  9.7, 10.5, 13. , 17.3, 18.1, 10.6,
        8.4,  9.1,  0. ,  6.8,  9.7, 14.6,  5. ,  9.1, 18. , 13.3,  8.6,
       10.8,  0. , 12.4, 13.9, 11.1])

array([ 8.7, 12.2, 10.4, 12.7, 12. , 10.6,  6.5,  6.1,  9. , 15.6,  0. ,
        0. ,  9.5,  8.1,  0. ,  9.9, 10.3, 14.1,  9.3, 10.1, 13.5, 11. ,
       11.4, 10.5,  6.8, 12.2,  0. , 13.8,  7.4,  0. ,  0. ,  0. , 12.9,
        0. ,  7.3,  8.2,  0. ,  8.8,  9.7, 10.5, 13. , 17.3, 18.1, 10.6,
        8.4,  9.1,  0. ,  6.8,  9.7, 14.6,  5. ,  9.1, 18. , 13.3,  8.6,
       10.8,  0. , 12.4, 13.9, 11.1])

In [19]:
#BUILD THE MODEL
# define cnn model

model = keras.Sequential([
    #keras.layers.Conv2D(32, kernel_size = (3, 3), activation='relu', kernel_initializer='he_uniform', input_shape=(28, 28, 1)),
    #keras.layers.MaxPooling2D( pool_size = (2, 2)),
    #keras.layers.Flatten(),
    keras.layers.Dense(139, activation='sigmoid'),
    keras.layers.Dense(100, activation='sigmoid'),
    keras.layers.Dense(29, activation='softmax')
])

# compile model
#opt = SGD(learning_rate=0.01, momentum=0.9)
model.compile(optimizer="adam", loss='sparse_categorical_crossentropy', metrics=['accuracy'])


In [20]:
 model.fit(train_x, train_y, epochs = 10)

Train on 139 samples
Epoch 1/10
139/139 [==============================] - 0s 2ms/sample - loss: nan - accuracy: 0.0719
Epoch 2/10
139/139 [==============================] - 0s 195us/sample - loss: nan - accuracy: 0.1295
Epoch 3/10
139/139 [==============================] - 0s 178us/sample - loss: nan - accuracy: 0.1295
Epoch 4/10
139/139 [==============================] - 0s 194us/sample - loss: nan - accuracy: 0.1295
Epoch 5/10
139/139 [==============================] - 0s 188us/sample - loss: nan - accuracy: 0.1295
Epoch 6/10
139/139 [==============================] - 0s 188us/sample - loss: nan - accuracy: 0.1295
Epoch 7/10
139/139 [==============================] - 0s 206us/sample - loss: nan - accuracy: 0.1295
Epoch 8/10
139/139 [==============================] - 0s 184us/sample - loss: nan - accuracy: 0.1295
Epoch 9/10
139/139 [==============================] - 0s 193us/sample - loss: nan - accuracy: 0.1295
Epoch 10/10
139/139 [==============================] - 0s 190us/sample -

In [10]:
test_loss, test_acc=model.evaluate(validation_x,validation_y)
print("Testing set loss :  %.3f"%test_loss)
#Eout.append((100-(100*test_acc)))
print("Testing set error :  %.2f"%(100-(100*test_acc)), "%")

60/1 [==================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================

In [63]:
#what is kernel_initalizer
    #normal distribution
#what does it mean to normalize data
#how do i make the correct input_shape
#what do all the different activation types do?
#what are the different optimizers
#how do I change the loss types?
    #what are the different loss types?
#what is metrics?

    keras.layers.Dense(100, activation='sigmoid'),
    keras.layers.Dense(80, activation='sigmoid'),
    keras.layers.Dense(19, activation='softmax') 
    15 epochs
    0.1295